Golden: Add weight to Pet entity

fmagoge opened the pull request:
-Added weight field
-Updated schema
-Added test cases

fmagoge committed:
Golden commit

-Added weight field
-Updated schema
-Added test cases

DmitriyChernyakSetronica commented:
FAIL_TO_PASS fields are not filled in.

fmagoge committed:
Petweight  Functionality tests

FAIL_TO_PASS , PASS_TO_PASS tests.

fmagoge commented:
@DmitriyChernyakSetronica I included the tests

Gowrishbyndoor commented on src/main/java/org/springframework/samples/petclinic/rest/controller/OwnerRestController.java:
```diff
@@ -156,6 +157,9 @@ public ResponseEntity<Void> updateOwnersPet(Integer ownerId, Integer petId, PetF
             if (currentPet != null) {
                 currentPet.setBirthDate(petFieldsDto.getBirthDate());
                 currentPet.setName(petFieldsDto.getName());
+                if (petFieldsDto.getWeight() != null) {
```
as per the requirement weight should allow null vales, but if you add this check then it will not allow null in update

![Uploading image.png…]()

Gowrishbyndoor commented on src/main/java/org/springframework/samples/petclinic/rest/controller/PetRestController.java:
```diff
@@ -79,6 +80,9 @@ public ResponseEntity<PetDto> updatePet(Integer petId, PetDto petDto) {
         }
         currentPet.setBirthDate(petDto.getBirthDate());
         currentPet.setName(petDto.getName());
+        if (petDto.getWeight() != null) {
```
same as above

Gowrishbyndoor commented on src/test/java/org/springframework/samples/petclinic/rest/controller/PetWeightFunctionalityTests.java:
```diff
@@ -0,0 +1,499 @@
+    }
+
+    @Nested
+    @DisplayName("FAIL_TO_PASS: Weight Field Requirements")
```
no need to mention here the FAIL_TO_PASS, just mention description about the tests should be enough in displayName. FAIL_TO_PASS should be available in your commit message and PR description.

Gowrishbyndoor commented on src/test/java/org/springframework/samples/petclinic/rest/controller/PetWeightFunctionalityTests.java:
```diff
@@ -0,0 +1,499 @@
+
+            String petJson = objectMapper.writeValueAsString(updatedPet);
+
+            mockMvc.perform(put("/api/pets/1")
```
After update, check the correct values saved in the DB using the GET api

Gowrishbyndoor commented on src/test/java/org/springframework/samples/petclinic/rest/controller/PetWeightFunctionalityTests.java:
```diff
@@ -0,0 +1,499 @@
+        @WithMockUser(roles = "OWNER_ADMIN")
+        @DisplayName("POST /api/owners/{ownerId}/pets should accept weight field when creating pets")
+        void createPetShouldAcceptWeightField() throws Exception {
+            given(clinicService.findOwnerById(1)).willReturn(testOwner);
```
please write pure integration tests without mocking for all the scenarios

Gowrishbyndoor commented on src/test/java/org/springframework/samples/petclinic/rest/controller/PetWeightFunctionalityTests.java:
```diff
@@ -0,0 +1,499 @@
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .content(petJson)
+                    .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated()); // FAIL_TO_PASS: Should create pet with weight
```
Even here after POST success, use GET api to check right values are inserted.

This comment is applicable for all test cases where ever applicable

Gowrishbyndoor commented on src/test/java/org/springframework/samples/petclinic/rest/controller/PetWeightFunctionalityTests.java:
```diff
@@ -0,0 +1,499 @@
+
+    @Nested
+    @DisplayName("PASS_TO_PASS: Backward Compatibility and Regression Tests")
+    class BackwardCompatibilityTests {
```
Do we really need this class here, we just need 2 different test files with 2 different types of API, Pet API and Owner API. Write the respective tests inside the respective files. Then each API will be covered with dirreferent integration scenario like 200/400/500. These tests should be applicable to your requirement only, no need to write different tests which is not part of the requirement.

Gowrishbyndoor reviewed (changes requested):
Please address these:

- Golden PR should contains only one commit and commit message should be task description and PASS_TO_PASS or FAIL_TO_PASS details. Agent with golden tests should have 2 commits.
- Write the pure integration test without mocking and without importing the Models.
- Follow the tests pattern mentioned in the other comment. Should have proper possible positive and negative scenario test cases and test should cover only mentioned in the requirement. Means take one example of POST api, you should write 200/400/500 status tests with respect to weight field only not with other mandatory fields.
- After addressing the all these update the agent with golden tests PR with latest tests.

fmagoge committed:
FAIL_TO_PASS and integrations tests

 - current pet weight can now be set to null
 - removed BackwardCompatibilityTests class
 - changed mock tests to integration tests

FAIL_TO_PASS:

PetWeightFunctionalityTests

fmagoge committed:
Tests

adjusted tests
